### PR TITLE
feat: include required input file command line arguments. Fix typo…

### PR DESCRIPTION
**Pull request by welozano**
Include required command line arguments for input files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,12 +6,12 @@ This utility matches ground truth data activities in excel format with activitie
...
-`python matchAndMerge.py`
+`python matchAndMerge.py --obaFile obaFile.csv --gtFile gtFile.xlsx`
 
-### Input Data
+### Required Input Data
 matchAndMerge works over two expected input data files, one generates from OBA firebase export and one containing the ground truth data:
-* **OBA data file** - A csv file generated by [OBA firebase export](https://github.com/CUTR-at-USF/onebusaway-firebase-export).
-* **Ground truth data** - A xlsx file that must be formatted as shown below. The main column descriptions are:
+* `--obaFile <oba csv file>` A csv file generated by [OBA firebase export](https://github.com/CUTR-at-USF/onebusaway-firebase-export).
+* `--gtFile <ground truth xlsx file>` A xlsx file that must be formatted as shown below. The main column descriptions are:
```
Comment (barbeau) [inline README.md]: Re: the modes listed in line 16 below - The modes we get directly from Android are here:
https://developer.android.com/guide/topics/location/transitions

We add one more IIRC for "WALKING/RUNNING" (or some similar combination of the two) in the export tool when we combine multiple walk/run events.

So right now the OBA data for bus trips will be labeled `IN_VEHICLE`, but in the ground truth data we obviously need `IN_BUS` for bus trips. And we need similar labels for bikeshare (maybe? Or we could just label these as `ON_BICYCLE`).

Could you please add the definitive, exhaustive list of modes that appear either in OBA or in the ground truth data here in line 16 so we have a record of them?

File: src/gtMerger/matchAndMerge.py [hunk 2]
```diff
@@ -28,60 +29,78 @@
...
+
     # Create sub-folders for output an logs
-    path_logs = os.path.join(args.dataDir, "logs")
+    path_logs = os.path.join(command_line_args.dataDir, "logs")
     if not os.path.isdir(path_logs):
-        os.mkdir(path_logs)
+        try:
+            os.mkdir(path_logs)
+        except OSError:
+            print("There was an error while creating the sub folder for logs:", path_logs)
```
Comment (barbeau) [inline src/gtMerger/matchAndMerge.py:58]: Nit - should be `sub-folder` throughout.

File: src/gtMerger/matchAndMerge.py [hunk 3]
```diff
@@ -95,28 +114,28 @@ def preprocessObaData(data_csv, args):
         subset=['Activity Start Date and Time* (UTC)', 'Origin location Date and Time (*best) (UTC)',
                 'Duration* (minutes)', 'Origin-Destination Bird-Eye Distance* (meters)'])
 
-    # Remove trips with Duration less than args.minActivityDuration minutes and distance les than args.minTripLength
-    clean_data_csv = clean_data_csv[(data_csv['Duration* (minutes)'] >= args.minActivityDuration) & (
-                data_csv['Origin-Destination Bird-Eye Distance* (meters)'] >= args.minTripLength)]
+    # Remove trips with Duration less than command_line_args.minActivityDuration minutes and distance les than
```
Comment (barbeau) [inline src/gtMerger/matchAndMerge.py]: Typo "les than"

File: src/gtMerger/matchAndMerge.py [hunk 4]
```diff
@@ -95,28 +114,28 @@ def preprocessObaData(data_csv, args):
...
         '_merge', axis=1).reset_index(drop=True)
 
     # Save data to be dropped to a csv file
-    dropped_file_path = os.path.join(args.dataDir, "logs", "droppedObaData.csv")
+    dropped_file_path = os.path.join(command_line_args.dataDir, "logs", "droppedObaData.csv")
     data_csv_dropped.to_csv(path_or_buf=dropped_file_path, index=False)
 
     # Add column to be used in function "merge_asoft"
-    clean_data_csv['ClosestTime'] = clean_data_csv['Activity Start Date and Time* (UTC)']
+    # clean_data_csv['ClosestTime'] = clean_data_csv['Activity Start Date and Time* (UTC)']
```
Comment (barbeau) [inline src/gtMerger/matchAndMerge.py]: Can this commented-out code be removed? If not, you should probably explain why it's commented out and should be kept.

File: src/gtMerger/matchAndMerge.py [hunk 5]
```diff
@@ -164,6 +182,36 @@ def preprocessGtData(gt_data):
...
+    list_collectors = gt_data['GT_Collector'].unique()
+    list_oba_users = oba_data['User ID'].unique()
+
+    merged_df = pd.DataFrame()
+    for collector in list_collectors:
+        print("Merging data for collector ", collector)
+        # Create dataframe for a collector on list_collectors
+        gt_data_collector = gt_data[gt_data["GT_Collector"] == collector]
+        for oba_user in list_oba_users:
+            print("\t Oba user", oba_user)
```
Comment (barbeau) [inline src/gtMerger/matchAndMerge.py]: Does this include the OBA data user ID? If so, let's print an simple counting integer instead, or truncate the user ID to the last 4 digits if you need to see the ID. I'd prefer not to leak user IDs in log output.

**Review by barbeau — COMMENTED** [review]
Thanks @welozano, some comments in-line.

File: README.md (hunk 1 above)
Comment (welozano) [reply README.md]: I added the detected activity (mode) current list for OBA along with the description of other required columns in the csv file.

File: src/gtMerger/matchAndMerge.py (hunk 2 above)
Comment (welozano) [reply src/gtMerger/matchAndMerge.py:58]: Done.

**Review by barbeau — APPROVED** [review]
Thanks @welozano!

**Pull request merged**